src/main.rs

```diff
@@ -1278,17 +1282,26 @@ fn event_handler(data: &[u8]) -> ::std::os::raw::c_int {
     .unwrap()
     .elapsed();
 
-    let size = event.length as usize;
-    cprintln!(
-        Styles::Bpf,
-        "B: {:06}.{:06} {} {}",
-        elapsed.as_secs(),
-        elapsed.as_micros() % 1000000,
-        event.length,
-        event.data[..size]
-            .iter()
-            .fold("".to_string(), |acc, b| format!("{acc}{b:02x} "))
-    );
+    let size = if event.packet_number == event.packet_count - 1 {
+        event.length as usize - event.packet_number as usize * PACKET_SIZE
+    } else {
+        PACKET_SIZE
+    };
+    let bytes = event.data[..size]
+        .iter()
+        .fold("".to_string(), |acc, b| format!("{acc}{b:02x} "));
+    if event.packet_number == 0 {
+        cprintln!(
+            Styles::Bpf,
+            "B: {:06}.{:06} {} {}",
+            elapsed.as_secs(),
+            elapsed.as_micros() % 1000000,
+            event.length,
+            bytes,
+        );
+    } else {
+        cprintln!(Styles::Bpf, "B:                  {}", bytes,);
```

Comment:
> We have the packet_number in the struct, why don't we collect all packets and print them all as one line?

that would make the producer harder. We are in a callback, so we need some sort of static buffer, and these are notably hard to do in rust. I'll see if I can pull it from my hat.

> Make the number count down so it indicates remaining packets and then we know when to stop and print the \n.

So you mean use `cprint!()`, not `cprintln!()`? What if we get a partial bpf event and then move to hidraw?

Using the remaining packets can be already computed because I have the packet count and the packet number in the event, so it's just an difference away.

Reply:
So I could just use a:
```
let mut bpf_vec = Vec::new();
...
        builder.add(map_ringbuf, |data| event_handler(data, &mut bpf_vec)).unwrap();
```

And in the `event_handler()` call, I have this shared buffer available :)

So no interaction between hidraw and BPF, and we are good